### PR TITLE
[WFLY-19610]  @PostConstruct on Servlet may be called twice

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-19610

- Attempt to prevent double post construct invocation for injection targets that are beans
- Add reproducer/test

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/wildfly/test/integration/web/postcontstruct/beanjar/SimpleBean.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package org.wildfly.test.integration.web.postcontstruct.beanjar;
+
+public class SimpleBean {
```
Comment (manovotn) [inline testsuite/integration/basic/src/test/java/org/wildfly/test/integration/web/postcontstruct/beanjar/SimpleBean.java:3]: Nitpickers note: I'd add a bean defining annotation to this class (`@Dependent`/`@ApplicationScoped`/...).

**Review by manovotn — APPROVED** [review]

File: testsuite/integration/basic/src/test/java/org/wildfly/test/integration/web/postcontstruct/beanjar/SimpleBean.java (hunk 1 above)
Comment (jasondlee) [reply testsuite/integration/basic/src/test/java/org/wildfly/test/integration/web/postcontstruct/beanjar/SimpleBean.java:3]: I probably would, too, but this is what the reporter provided in the reproducer, so I've tried to change that as little as possible.

**Review by bstansberry — CHANGES_REQUESTED** [review]
Please add the headers to all files.

**Comment by jasondlee** [comment]
> Please add the headers to all files.

Done.

**Review by bstansberry — APPROVED** [review]

File: testsuite/integration/basic/src/test/java/org/wildfly/test/integration/web/postcontstruct/beanjar/SimpleBean.java (hunk 1 above)
Comment (manovotn) [reply testsuite/integration/basic/src/test/java/org/wildfly/test/integration/web/postcontstruct/beanjar/SimpleBean.java:8]: It makes no difference for the purpose of this test - it's just that you now rely on having `beans.xml` with explicitly stated `all` discovery mode whereas with added annotation, the test would work even with empty `beans.xml` or completely without it :)

**Review by manovotn — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @jasondlee and @manovotn

**Pull request merged**